cypress/integration/detectors_spec.js

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  PLUGIN_NAME,
+  TWENTY_SECONDS_TIMEOUT,
+  TEST_INDEX,
+  TEST_FIELD_MAPPINGS,
+} from '../support/constants';
+
+describe('Detectors', () => {
+  before(() => {
+    cy.deleteAllIndices();
+
+    // Create test index
+    cy.createIndex('cypress-test-windows', TEST_INDEX);
+
+    cy.contains('test detector').should('not.exist');
+  });
+
+  beforeEach(() => {
+    // Visit Detectors page
+    cy.visit(`${Cypress.env('opensearch_dashboards')}/app/${PLUGIN_NAME}#/detectors`);
+    //wait for page to load
+    cy.wait(10000);
+
+    // Check that correct page is showing
+    cy.contains('Threat detectors');
+  });
+
+  it('...can be created', () => {
+    // Locate Create detector button click to start
+    cy.contains('Create detector').click({ force: true });
+
+    // Check to ensure process started
+    cy.contains('Define detector');
+
+    // Enter a name for the detector in the appropriate input
+    cy.get(`input[placeholder="Enter a name for the detector."]`).type('test detector{enter}');
+
+    // Select our pre-seeded data source (cypress-test-windows)
+    cy.get(`[data-test-subj="define-detector-select-data-source"]`).type(
+      'cypress-test-windows{enter}'
+    );
+
+    // Select threat detector type (Windows logs)
+    cy.get(`input[id="windows"]`).click({ force: true });
+
+    // Wait for detector rules to load - timeout on click above ineffective
+    cy.wait(10000);
+
+    // Click Next button to continue
+    cy.get('button').contains('Next').click({ force: true }, { timeout: 2000 });
+
+    // Check that correct page now showing
+    cy.contains('Required field mappings');
+
+    // Select appropriate names to map fields to
+    for (let field_name in TEST_FIELD_MAPPINGS) {
+      const mappedTo = TEST_FIELD_MAPPINGS[field_name];
+
+      cy.contains('tr', field_name).within(() => {
+        cy.get(`[data-test-subj="detector-field-mappins-select"]`).select(mappedTo);
+      });
+    }
```

Comment:
This might not work correctly since we don't get the mappings as per the Test mappings